pkg/lib/ssh-add-key.sh

```diff
@@ -0,0 +1,19 @@
+#! /bin/sh
+
+set -eufx
+
+d=$HOME/.ssh
+f=$d/authorized_keys
+
+test -f "$f" || ( mkdir -p "$d" && chmod 700 "$d" && touch "$f" && chmod 600 "$f" )
```

Comment:
No need for a subshell here. Use { } or just if/then.

Also, use `mkdir -m 700` -- if the ~/.ssh directory already exists, you don't want to change its permissions. Likewise with the chmod 600 -- if the authorized_keys file already exists, don't change it. For both it may just be better to set `umask 077` in the script.

Reply:
> if the authorized_keys file already exists, don't change it.

This isn't done; nothing is changed when `test -f` is true.

Reply:
Ah right, so just the `mkdir -m 700` or the umask. The file perms are ok indeed, sorry.

Reply:
Done